api/src/services/assets.ts

```diff
@@ -178,7 +180,20 @@ export class AssetsService {
 				readStream.unpipe(transformer);
 			});
 
-			await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			try {
+				await storage.location(file.storage).write(assetFilename, readStream.pipe(transformer), type);
+			} catch (error) {
+				await storage.location(file.storage).delete(assetFilename);
+
+				if ((error as Error)?.message?.includes('timeout')) {
+					throw new ServiceUnavailableError({
+						service: 'asset',
```

Comment:
```suggestion
						service: 'assets',
```

Little attempt at consistency, though it's an assumption based on how `files` was used instead of `file` elsewhere:

https://github.com/directus/directus/blob/523c57739cedc5795b3ea2c7f2c9471aef933988/api/src/services/assets.ts#L154-L157